ultranest/popstepsampler.py

```diff
@@ -688,7 +688,333 @@ def __next__(
             return None, None, None, nc
 
 
+
+
+def slice_limit_to_unitcube(tleft, tright):
+    
+    """
+    return the slice limits as a copy of intersection between the slice and the unit cube boundaries
```

Comment:
`Returns a copy of the given slice limits.

The passed slice limits are assumed to already be coming from the intersection of the unit cube.
`

remove mention of copy here, it is a technical detail.